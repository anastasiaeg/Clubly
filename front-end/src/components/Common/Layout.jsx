import React from "react";
import { Link } from "react-router-dom";
import {
  Alignment,
  Button,
  Card,
  Elevation,
  NavbarDivider,
  NavbarGroup
} from "@blueprintjs/core";
import "./common.css";
import * as routes from "../../constants/routes";
import { authService } from "../../services/authService";

class Layout extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      height: window.innerHeight,
      width: window.innerWidth
    };
  }

  updateDimensions = () => {
    this.setState({
      height: window.innerHeight,
      width: window.innerWidth
    });
  };

  componentWillMount() {
    this.updateDimensions();
  }

  componentDidMount() {
    window.addEventListener("resize", this.updateDimensions);
    this.updateDimensions();
  }

  componentWillUnmount() {
    window.removeEventListener("resize", this.updateDimensions);
  }

  onLogout = () => {
    authService.logout();
  };

  navbar = () => {
    const { route, user } = this.props;
    const location = route.location.pathname.split("/", 2)[1];

    if (user.type === "admin") {
      return (
        <NavbarGroup align={Alignment.LEFT} className="dont-highlight">
          <Link to={routes.ADMIN}>
            <Button
              className="bp3-minimal"
              icon="helper-management"
              text="Admin"
              large={true}
              active={location === "admin" || location === "manage_admin"}
            />
          </Link>
        </NavbarGroup>
      );
    }

    return (
      <NavbarGroup align={Alignment.LEFT} className="dont-highlight">
        <Link to={routes.PROFILE}>
          <Button
            className="bp3-minimal"
            icon="user"
            text="Profile"
            large={true}
            active={location === "profile"}
          />
        </Link>
        <NavbarDivider />
        <Link to={routes.CALENDAR}>
          <Button
            className="bp3-minimal"
            icon="calendar"
            text="Calendar"
            large={true}
            active={location === "" || location === "calendar"}
          />
        </Link>
        <NavbarDivider />
        <Link to={routes.SEARCH}>
          <Button
            className="bp3-minimal"
            icon="search"
            text="Search"
            large={true}
            active={location === "search"}
          />
        </Link>
        {user.type === "organizer" && <NavbarDivider /> && (
            <Link to={routes.MANAGE_ORGS}>
              <Button
                className="bp3-minimal"
                icon="helper-management"
                text="Manage Orgs"
                large={true}
                active={location === "org" || location === "manageOrgs"}
              />
            </Link>
          )}
      </NavbarGroup>
    );
  };

  render() {
    const { route, user } = this.props;
    const { height, width } = this.state;
    const Page = this.props.page;
    let navbarElement = this.navbar();
    return (
      <div className="layout">
        <nav className="bp3-navbar fixedToTop">
          <div
            className="center-horizontal"
            style={{
              width:
                user.type === "admin"
                  ? 100
                  : user.type === "organizer"
                    ? 600
                    : 400
            }}
          >
            <NavbarGroup align={Alignment.LEFT} className="dont-highlight">
              {navbarElement}
            </NavbarGroup>
          </div>
          <NavbarGroup align={Alignment.RIGHT}>
            <Link to={routes.LOGIN}>
              <Button
                className="bp3-minimal"
                text="Logout"
                large={true}
                onClick={this.onLogout}
              />
            </Link>
          </NavbarGroup>
        </nav>
        <Card
          elevation={Elevation.TWO}
          className="card"
          style={{ height: height - 85 + "px" }}
        >
          <Page route={route} height={height} width={width} user={user} />
        </Card>
      </div>
    );
  }
}

export default Layout;
