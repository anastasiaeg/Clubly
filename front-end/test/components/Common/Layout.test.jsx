import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";

import Layout from "../../../src/components/Common/Layout";
import TestComponent from "./TestComponent";

const params = {
  route: {
    location: {
      pathname: "/calendar"
    }
  }
};

describe("Layout", () => {
  test("should render layout", () => {
    const component = renderer
      .create(
        <MemoryRouter>
          <Layout route={params.route} page={TestComponent} />
        </MemoryRouter>
      )
      .toJSON();

    var navGroup = component.children[0].children[0].children[0].children[0];
    expect(navGroup.children[0].type).toBe("a");
    expect(navGroup.children[0].props.href).toBe("/profile");
    expect(navGroup.children[1].props.className).toBe("bp3-navbar-divider");
    expect(navGroup.children[2].type).toBe("a");
    expect(navGroup.children[2].props.href).toBe("/calendar");
    expect(navGroup.children[3].props.className).toBe("bp3-navbar-divider");
    expect(navGroup.children[4].type).toBe("a");
    expect(navGroup.children[4].props.href).toBe("/search");
    expect(navGroup.children[5].props.className).toBe("bp3-navbar-divider");
    expect(navGroup.children[6].type).toBe("a");
    expect(navGroup.children[6].props.href).toBe("/manageOrgs");

    expect(component).toMatchSnapshot();
  });
});
