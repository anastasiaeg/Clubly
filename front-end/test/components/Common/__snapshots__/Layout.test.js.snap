// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render 1`] = `
<div
  className="layout"
>
  <nav
    className="bp3-navbar fixedToTop"
  >
    <div
      className="center-horizontal"
      style={
        Object {
          "width": 400,
        }
      }
    >
      <div
        className="bp3-navbar-group bp3-align-left dont-highlight"
      >
        <div
          className="bp3-navbar-group bp3-align-left dont-highlight"
        >
          <a
            href="/profile"
            onClick={[Function]}
          >
            <button
              className="bp3-button bp3-large bp3-minimal"
              disabled={undefined}
              onClick={undefined}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              tabIndex={undefined}
              type="button"
            >
              <span
                className="bp3-icon bp3-icon-user"
                icon="user"
              >
                <svg
                  data-icon="user"
                  fill={undefined}
                  height={16}
                  viewBox="0 0 16 16"
                  width={16}
                >
                  <desc>
                    user
                  </desc>
                  <path
                    d="M7.99-.01A7.998 7.998 0 0 0 .03 8.77c.01.09.03.18.04.28.02.15.04.31.07.47.02.11.05.22.08.34.03.13.06.26.1.38.04.12.08.25.12.37.04.11.08.21.12.32a6.583 6.583 0 0 0 .3.65c.07.14.14.27.22.4.04.07.08.13.12.2l.27.42.1.13a7.973 7.973 0 0 0 3.83 2.82c.03.01.05.02.07.03.37.12.75.22 1.14.29l.2.03c.39.06.79.1 1.2.1s.81-.04 1.2-.1l.2-.03c.39-.07.77-.16 1.14-.29.03-.01.05-.02.07-.03a8.037 8.037 0 0 0 3.83-2.82c.03-.04.06-.08.09-.13.1-.14.19-.28.28-.42.04-.07.08-.13.12-.2.08-.13.15-.27.22-.41.04-.08.08-.17.12-.26.06-.13.11-.26.17-.39.04-.1.08-.21.12-.32.04-.12.08-.24.12-.37.04-.13.07-.25.1-.38.03-.11.06-.22.08-.34.03-.16.05-.31.07-.47.01-.09.03-.18.04-.28.02-.26.04-.51.04-.78-.03-4.41-3.61-7.99-8.03-7.99zm0 14.4c-1.98 0-3.75-.9-4.92-2.31.67-.36 1.49-.66 2.14-.95 1.16-.52 1.04-.84 1.08-1.27.01-.06.01-.11.01-.17-.41-.36-.74-.86-.96-1.44v-.01c0-.01-.01-.02-.01-.02-.05-.13-.09-.26-.12-.39-.28-.05-.44-.35-.5-.63-.06-.11-.18-.38-.15-.69.04-.41.2-.59.38-.67v-.06c0-.51.05-1.24.14-1.72.02-.13.05-.26.09-.39.17-.59.53-1.12 1.01-1.49.49-.38 1.19-.59 1.82-.59.62 0 1.32.2 1.82.59.48.37.84.9 1.01 1.49.04.13.07.26.09.4.09.48.14 1.21.14 1.72v.07c.18.08.33.26.37.66.03.31-.1.58-.16.69-.06.27-.21.57-.48.62-.03.13-.07.26-.12.38 0 .01-.01.04-.01.04-.21.57-.54 1.06-.94 1.42 0 .06.01.13.01.19.04.43-.12.75 1.05 1.27.65.29 1.47.6 2.14.95a6.415 6.415 0 0 1-4.93 2.31z"
                    fillRule="evenodd"
                  />
                </svg>
              </span>
              <span
                className="bp3-button-text"
              >
                Profile
              </span>
            </button>
          </a>
          <div
            className="bp3-navbar-divider"
          />
          <a
            href="/calendar"
            onClick={[Function]}
          >
            <button
              className="bp3-button bp3-active bp3-large bp3-minimal"
              disabled={undefined}
              onClick={undefined}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              tabIndex={undefined}
              type="button"
            >
              <span
                className="bp3-icon bp3-icon-calendar"
                icon="calendar"
              >
                <svg
                  data-icon="calendar"
                  fill={undefined}
                  height={16}
                  viewBox="0 0 16 16"
                  width={16}
                >
                  <desc>
                    calendar
                  </desc>
                  <path
                    d="M11 3c.6 0 1-.5 1-1V1c0-.6-.4-1-1-1s-1 .4-1 1v1c0 .5.4 1 1 1zm3-2h-1v1c0 1.1-.9 2-2 2s-2-.9-2-2V1H6v1c0 1.1-.9 2-2 2s-2-.9-2-2V1H1c-.6 0-1 .5-1 1v12c0 .6.4 1 1 1h13c.6 0 1-.4 1-1V2c0-.6-.5-1-1-1zM5 13H2v-3h3v3zm0-4H2V6h3v3zm4 4H6v-3h3v3zm0-4H6V6h3v3zm4 4h-3v-3h3v3zm0-4h-3V6h3v3zM4 3c.6 0 1-.5 1-1V1c0-.6-.4-1-1-1S3 .4 3 1v1c0 .5.4 1 1 1z"
                    fillRule="evenodd"
                  />
                </svg>
              </span>
              <span
                className="bp3-button-text"
              >
                Calendar
              </span>
            </button>
          </a>
          <div
            className="bp3-navbar-divider"
          />
          <a
            href="/search"
            onClick={[Function]}
          >
            <button
              className="bp3-button bp3-large bp3-minimal"
              disabled={undefined}
              onClick={undefined}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              tabIndex={undefined}
              type="button"
            >
              <span
                className="bp3-icon bp3-icon-search"
                icon="search"
              >
                <svg
                  data-icon="search"
                  fill={undefined}
                  height={16}
                  viewBox="0 0 16 16"
                  width={16}
                >
                  <desc>
                    search
                  </desc>
                  <path
                    d="M15.55 13.43l-2.67-2.68a6.94 6.94 0 0 0 1.11-3.76c0-3.87-3.13-7-7-7s-7 3.13-7 7 3.13 7 7 7c1.39 0 2.68-.42 3.76-1.11l2.68 2.67a1.498 1.498 0 1 0 2.12-2.12zm-8.56-1.44c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
                    fillRule="evenodd"
                  />
                </svg>
              </span>
              <span
                className="bp3-button-text"
              >
                Search
              </span>
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      className="bp3-navbar-group bp3-align-right"
    >
      <a
        href="/manageOrgs"
        onClick={[Function]}
      >
        <button
          className="bp3-button bp3-large bp3-minimal"
          disabled={undefined}
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          tabIndex={undefined}
          type="button"
        >
          <span
            className="bp3-button-text"
          >
            Logout
          </span>
        </button>
      </a>
    </div>
  </nav>
  <div
    className="bp3-card bp3-elevation-2 card"
    style={
      Object {
        "height": "683px",
      }
    }
  >
    <div
      className="flex-center"
    >
      HELLO
    </div>
  </div>
</div>
`;
